pkg/apis/networking/v1alpha1/certificate_types.go

```diff
@@ -0,0 +1,139 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/knative/pkg/apis"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Certificate is responsible for provisioning a SSL certificate for the
+// given hosts. It is a Knative abstraction for various SSL certificate
+// provisioning solutions (such as cert-manager or self-signed SSL certificate).
+type Certificate struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the desired state of the Certificate.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec CertificateSpec `json:"spec,omitempty"`
+
+	// Status is the current state of the Certificate.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status CertificateStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CertificateList is a collection of `Certificate`.
+type CertificateList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	// Items is the list of `Certificate`.
+	Items []Certificate `json:"items"`
+}
+
+// CertificateSpec defines the desired state of Certificate.
+type CertificateSpec struct {
+	// DNSNames is a list of DNS names the Certificate could support.
+	DNSNames []string `json:"dnsNames"`
+
+	// SecretName is the name of the secret resource to store the SSL certificate in.
+	SecretName string `json:"secretName"`
+}
+
+// CertificateStatus defines the observed state of Certificate.
+type CertificateStatus struct {
+	// The detailed information related to the provisioned certificate.
+	// +optional
+	CertificateInfo CertificateInfo `json:"certificateInfo,omitempty"`
+
+	// +optional
+	Conditions duckv1alpha1.Conditions `json:"conditions,omitempty"`
+}
+
+// InitializeConditions initializes the certificate conditions.
+func (cs *CertificateStatus) InitializeConditions() {
+	certificateCondSet.Manage(cs).InitializeConditions()
+}
+
+// MarkReady marks the certificate as ready to use.
+func (cs *CertificateStatus) MarkReady() {
+	certificateCondSet.Manage(cs).MarkTrue(CertificateCondidtionReady)
+}
+
+// GetConditions returns the Conditions array. This enables generic handling of
```

Comment:
See https://github.com/knative/serving/pull/3441, we no longer need these methods.